tests/support/server.rs

```diff
@@ -40,41 +41,52 @@ where
     F: Fn(http::Request<hyper::Body>) -> Fut + Clone + Send + 'static,
     Fut: Future<Output = http::Response<hyper::Body>> + Send + 'static,
 {
-    let srv = hyper::Server::bind(&([127, 0, 0, 1], 0).into()).serve(
-        hyper::service::make_service_fn(move |_| {
-            let func = func.clone();
-            async move {
-                Ok::<_, Infallible>(hyper::service::service_fn(move |req| {
-                    let fut = func(req);
-                    async move { Ok::<_, Infallible>(fut.await) }
-                }))
-            }
-        }),
-    );
+    //Spawn new runtime in thread to prevent reactor execution context conflict
+    thread::spawn(move || {
```

Comment:
not sure if there is a better solution for executors conflict. Could we just use `.await` here and execute on the default runtime?